p2p/simulations/adapters/inproc_test.go

```diff
@@ -25,22 +25,29 @@ import (
 )
 
 func TestSocketPipe(t *testing.T) {
```

Comment:
These tests expected a certain buffer size on `socketPipe`, which is not possible on Travis. Therefore I change the tests to be asynchronous, similarly to the `sim` adapter, and make no assumptions on the size of the OS buffer.